backend/data-preparation/data_collector.py

```diff
@@ -0,0 +1,159 @@
+import os
+import time
+from typing import List, Dict, Optional
+import logging
+from datetime import datetime
+from dotenv import load_dotenv
+
+from langchain_community.document_loaders.git import GitLoader
+from langchain_core.documents import Document
+
+from .mongo_setup import NigerianLawsDatabase
+
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+logger = logging.getLogger(__name__)
+
+load_dotenv()
+
+MONGO_URI = os.getenv("MONGO_URI")
+MONGO_DB_NAME = os.getenv("MONGO_DB_NAME")
+MONGO_COLLECTION_NAME = os.getenv("MONGO_COLLECTION_NAME") 
+GITHUB_PERSONAL_ACCESS_TOKEN = os.getenv("GITHUB_PERSONAL_ACCESS_TOKEN")
+
+TARGET_GITHUB_REPO_URL: str = "https://github.com/mykeels/nigerian-laws?tab=readme-ov-file"
```

Comment:
no need for the tab section
```suggestion
TARGET_GITHUB_REPO_URL: str = "https://github.com/mykeels/nigerian-laws"
```

Reply:
Instead of reading from website, another option is to clone the repository into your own and just read the content from there